analyzers/src/SonarAnalyzer.VisualBasic/Rules/WeakSslTlsProtocols.cs

```diff
@@ -34,11 +34,34 @@ public sealed class WeakSslTlsProtocols : WeakSslTlsProtocolsBase<SyntaxKind, Id
     {
         protected override GeneratedCodeRecognizer GeneratedCodeRecognizer { get; } = VisualBasicGeneratedCodeRecognizer.Instance;
 
-        protected override SyntaxKind SyntaxKind { get; } = SyntaxKind.IdentifierName;
+        protected override SyntaxKind SyntaxKindOfInterest { get; } = SyntaxKind.IdentifierName;
 
         protected override DiagnosticDescriptor Rule { get; } = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
 
         protected override string GetIdentifierText(IdentifierNameSyntax identifierNameSyntax) =>
             identifierNameSyntax.Identifier.Text;
+
+        protected override bool IsNodeOfInterest(SyntaxNode node)
+        {
+            if (!(node.Parent is MemberAccessExpressionSyntax))
+            {
+                return false;
+            }
+
+            var current = node;
+            var parent = current.Parent;
+            while (parent != null && !parent.IsKind(SyntaxKind.IfStatement))
+            {
+                current = parent;
+                parent = parent.Parent;
+            }
+
+            if (parent != null && parent.IsKind(SyntaxKind.IfStatement))
+            {
+                return (parent as IfStatementSyntax).Condition != current;
```

Comment:
I would use a direct cast here since we check the syntax kind previously and if the cast fails we will get a NRE anyway.

Reply:
You mean that  I should remove the if and just do a cast?

Reply:
No, I would replace the `as` with a direct cast.
```suggestion
                return ((IfStatementSyntax)parent).Condition != current;
```